### DISCUSSION TITLE
`rand()` in stdlib

### DISCUSSION THREAD

**Discussion by jessfraz** [KittyCAD/modeling-app#646] [New Feature Requests]
random integer etc

https://discord.com/channels/915388055236509727/1153710304656633976/1153794842015584326

**Comment by Irev-Dev** [comment]
We should only support `rand(seed)` in my opinion.

It's really important that we have deterministic scripts in part because we want to query both the ast and the program memory after an initial execution, and having things be consistent between executions is important.
As well as the fact that we want to support partial execution, and caching, having deterministic scripts a good idea.

It's possible that we could support a true rand with the above, but it would add a lot of complexity that I simply don't think is worth it for us now, so I think if we hand rand it has to be with a seed, and that goes for any other non-deterministic behaviour. I have a strong conviction on this.

I might be wrong on this but I see the use of rand() to be mostly a creative util, doesn't have much use for mechanical design (tell me if I'm wrong) an exception might be noise for a surface, but that doesn't need to be different for each execution, so rand with seed still stands.

If users really want to output multiple parts that vary because of a random value, they can implement it themselves by looping in bash with the cli and inputting env-vars (not that we support env-vars yet). That way when using the app to develop scripts in our sandbox of an app, everything is deterministic, and users can inject their own random values outside of the app.

**Comment by jessfraz** [comment]
```
fn random(i) {
  x = 0.123456789
  return reduce([1..i], x, fn(i, acc) {
    r = 3.9999
    return r * acc * (1 - acc)
  })
}
```

**Reply by lf94** [reply]
FWIW this is called a logistic map or logistic equation.
